### PR TITLE
added thriftsync usage in the docs

### PR DESCRIPTION
**Pull request by anuptalwalkar**

### CODE REVIEW

**Comment by anuptalwalkar** [comment]
@lupie

File: examples/keyvalue/Makefile [hunk 1]
```diff
@@ -19,3 +19,6 @@ deps:
 
 	@echo "Installing thriftrw-plugin-thriftsync..."
 	$(ECHO_V)go install ../../modules/rpc/thriftrw-plugin-thriftsync
+
+thriftsync:
```
Comment (sectioneight) [inline examples/keyvalue/Makefile]: should this depend on the binary being installed?

File: modules/rpc/thriftrw-plugin-thriftsync/README.md [hunk 2]
```diff
@@ -3,17 +3,45 @@
  *.thrift file. With the use of thriftsync plugin, a user who needs to build a service should be able
  to auto generate the code and write service specific logic without worrying about underlying platform.
 
+## Usage
+Run thriftsync on any thrift file update to sync your handler code with the methods in the thrift file.
```
Comment (sectioneight) [inline modules/rpc/thriftrw-plugin-thriftsync/README.md]: backticks around the command to show it's a binary

File: modules/rpc/thriftrw-plugin-thriftsync/README.md [hunk 3]
```diff
@@ -3,17 +3,45 @@
...
+To run thriftsync, you need to install thriftsync from vendor and run thriftrw code generation command.
```
Comment (sectioneight) [inline modules/rpc/thriftrw-plugin-thriftsync/README.md]: this sentence is a bit awkward

File: modules/rpc/thriftrw-plugin-thriftsync/README.md [hunk 4]
```diff
@@ -3,17 +3,45 @@
...
+
+*Run thriftrw code genration with thriftsync*
+`thriftrw --plugin="thriftsync --yarpc-server=<Import path of the yarpc servicenameserver>" <thrift filepath>`
+
+Update your makefile with following lines and simply run `make thriftsync`
+*Update makefile*
+```
+deps:
+	@echo "Installing thriftrw..."
+	$(ECHO_V)go install ./vendor/go.uber.org/thriftrw
```
Comment (sectioneight) [inline modules/rpc/thriftrw-plugin-thriftsync/README.md]: i'd remove the echo v stuff -- we use it internally but for examples it just makes it less clear what the important stuff is

File: modules/rpc/thriftrw-plugin-thriftsync/README.md [hunk 5]
```diff
@@ -3,17 +3,44 @@
  *.thrift file. With the use of thriftsync plugin, a user who needs to build a service should be able
```
Comment (Lupie) [inline modules/rpc/thriftrw-plugin-thriftsync/README.md]: Can you add links to thriftsync and thriftrw? They're not easily Googlable.

Comment (Lupie) [inline modules/rpc/thriftrw-plugin-thriftsync/README.md]: - of thriftsync plugin --> of the thriftsync plugin
- a user who needs to build --> a user building

File: modules/rpc/thriftrw-plugin-thriftsync/README.md [hunk 6]
```diff
@@ -3,17 +3,44 @@
...
  to auto generate the code and write service specific logic without worrying about underlying platform.
```
Comment (Lupie) [inline modules/rpc/thriftrw-plugin-thriftsync/README.md]: - autogenerate
- service-specific
- the underlying platform

File: modules/rpc/thriftrw-plugin-thriftsync/README.md [hunk 7]
```diff
@@ -3,17 +3,44 @@
...
 
+## Usage
+Run `thriftsync` on any thrift file update to sync your handler code with the methods in the thrift file.
```
Comment (Lupie) [inline modules/rpc/thriftrw-plugin-thriftsync/README.md]: - thrift --> Thrift (throughout)
- Meaning is unclear. Run the command on a thrift file…to sync? Should we cut "update"? I think we're either missing words or we have too many.

File: modules/rpc/thriftrw-plugin-thriftsync/README.md [hunk 8]
```diff
@@ -3,17 +3,44 @@
...
+
+*Install thriftsync from vendor:*
+`go install ./vendor/go.uber.org/fx/modules/rpc/thriftrw-plugin-thriftsync`
+
+*Run thriftrw code genration with thriftsync*
+`thriftrw --plugin="thriftsync --yarpc-server=<Import path of the yarpc servicenameserver>" <thrift filepath>`
+
+Update your makefile with following lines and simply run `make thriftsync`
```
Comment (Lupie) [inline modules/rpc/thriftrw-plugin-thriftsync/README.md]: - with following --> with the following
- Delete "simply"
- lines and --> lines, and

File: modules/rpc/thriftrw-plugin-thriftsync/README.md [hunk 9]
```diff
@@ -3,17 +3,44 @@
...
+  @echo "Installing thriftrw-plugin-thriftsync..."
+  go install ./vendor/go.uber.org/fx/modules/rpc/thriftrw-plugin-thriftsync
+
+thriftsync: deps
+  thriftrw --plugin="thriftsync --yarpc-server=<Import path of the yarpc servicenameserver>" <thrift filepath>
+```
+
 ## Example
-Following examples show how thriftsync syncs handler code with the updated thrift file:
+Following examples show how `thriftsync` syncs handler code with the updated thrift file:
```
Comment (Lupie) [inline modules/rpc/thriftrw-plugin-thriftsync/README.md]: Following --> The following

File: modules/rpc/thriftrw-plugin-thriftsync/README.md [hunk 10]
```diff
@@ -3,17 +3,44 @@
...
+Run `thriftsync` on any thrift file update to sync your handler code with the methods in the thrift file.
+
+*Install thriftsync from vendor:*
+`go install ./vendor/go.uber.org/fx/modules/rpc/thriftrw-plugin-thriftsync`
+
+*Run thriftrw code genration with thriftsync*
+`thriftrw --plugin="thriftsync --yarpc-server=<Import path of the yarpc servicenameserver>" <thrift filepath>`
+
+Update your makefile with following lines and simply run `make thriftsync`
+*Update makefile*
```
Comment (Lupie) [inline modules/rpc/thriftrw-plugin-thriftsync/README.md]: The steps in italics don't render well. Are you OK with numbering them instead, with step 1 in line 9?

File: modules/rpc/thriftrw-plugin-thriftsync/README.md [hunk 11]
```diff
@@ -3,17 +3,44 @@
...
 **New handler generation**
 
 ```thrift
+testservice.thrift
 service TestService {
   string testFunction(1: string param)
 }
 ```
 
+*Run thriftsync*
```
Comment (Lupie) [inline modules/rpc/thriftrw-plugin-thriftsync/README.md]: Why is this instruction in here with all these examples?

File: modules/rpc/thriftrw-plugin-thriftsync/README.md (hunk 5 above)
Comment (anuptalwalkar) [inline modules/rpc/thriftrw-plugin-thriftsync/README.md]: thriftsync is the current project, I'll add link to thriftrw

File: modules/rpc/thriftrw-plugin-thriftsync/README.md (hunk 7 above)
Comment (anuptalwalkar) [inline modules/rpc/thriftrw-plugin-thriftsync/README.md]: reduced to "Run `thriftsync` to sync your handler code with the methods in the Thrift file."

File: modules/rpc/thriftrw-plugin-thriftsync/README.md (hunk 11 above)
Comment (anuptalwalkar) [inline modules/rpc/thriftrw-plugin-thriftsync/README.md]: removed

**Pull request merged**